site/src/components/CopyableValue/CopyableValue.tsx

```diff
@@ -1,35 +1,91 @@
-import Tooltip, { type TooltipProps } from "@mui/material/Tooltip";
+import {
+	Tooltip,
+	TooltipContent,
+	TooltipProvider,
+	TooltipTrigger,
+} from "components/Tooltip/Tooltip";
 import { useClickable } from "hooks/useClickable";
 import { useClipboard } from "hooks/useClipboard";
-import type { FC, HTMLAttributes } from "react";
+import { type FC, type HTMLAttributes, useState } from "react";
+import { cn } from "utils/cn";
+
+type TooltipSide = "top" | "right" | "bottom" | "left";
 
 interface CopyableValueProps extends HTMLAttributes<HTMLSpanElement> {
 	value: string;
-	placement?: TooltipProps["placement"];
-	PopperProps?: TooltipProps["PopperProps"];
+	side?: TooltipSide;
 }
 
 export const CopyableValue: FC<CopyableValueProps> = ({
 	value,
-	placement = "bottom-start",
-	PopperProps,
+	side = "bottom",
 	children,
+	className,
+	role,
+	tabIndex,
+	onClick,
+	onKeyDown,
+	onKeyUp,
 	...attrs
 }) => {
 	const { showCopiedSuccess, copyToClipboard } = useClipboard();
+	const [tooltipOpen, setTooltipOpen] = useState(false);
+	const [isFocused, setIsFocused] = useState(false);
 	const clickableProps = useClickable<HTMLSpanElement>(() => {
 		copyToClipboard(value);
+		setTooltipOpen(true);
 	});
 
 	return (
-		<Tooltip
-			title={showCopiedSuccess ? "Copied!" : "Click to copy"}
-			placement={placement}
-			PopperProps={PopperProps}
-		>
-			<span {...attrs} {...clickableProps} css={{ cursor: "pointer" }}>
-				{children}
-			</span>
-		</Tooltip>
+		<TooltipProvider delayDuration={100}>
+			<Tooltip
+				open={tooltipOpen}
+				onOpenChange={(next) => {
```

Comment:
```suggestion
				onOpenChange={(shouldBeOpen) => {
```

the name `next` just makes this really hard to reason about for me. if I substitute it for something like this in my head it makes it clearer.